map/framework.cpp

```diff
@@ -517,10 +517,16 @@ bool Framework::OnCountryFileDelete(storage::TCountryId const & countryId, stora
 
 void Framework::OnMapDeregistered(platform::LocalCountryFile const & localFile)
```

Comment:
А почему тогда не вставить в том месте, где уменьшается счетчик и вызывается Deregister, RunOnGUIThread ? Ведь тогда поведение было бы консистентное и предсказуемое для всех.

> On Mar 25, 2016, at 22:48, ygorshenin <notifications@github.com> wrote:
> 
> In map/framework.cpp <https://github.com/mapsme/omim/pull/2523#discussion_r57462481>:
> 
> > @@ -517,10 +517,16 @@ bool Framework::OnCountryFileDelete(storage::TCountryId const & countryId, stora
> >  
> >  void Framework::OnMapDeregistered(platform::LocalCountryFile const & localFile)
> Да, именно. MwmHandle создаются кем угодно, на каких угодно потоках. И если мы хотим удалить карту, а есть еще активные MwmHandle на эту карту, то физическое удаление карты откладывается, хотя из UI она и исчезает. После этого на карту уже нельзя взять новый MwmHandle, но старые, взятые, еще активны и валидны. Когда они уничтожаются со своих каких-то потоков, они уменьшают счетчик ссылок на один, и когда самый последний уничтожается, вызывается этот callback, с того потока на котором и был уничтожен последний MwmHandle. Это может быть какой угодно поток (роутинг, поиск, вспомогательные потоки...) поэтому гарантий что это будет главный поток нет никаких.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub <https://github.com/mapsme/omim/pull/2523/files/8bff1dbc61d252545ea4b21bffd137f33f43a0e2#r57462481>